pkg/component/autoscaling/vpa/vpa_test.go

```diff
@@ -1665,18 +1713,87 @@ var _ = Describe("VPA", func() {
 
 		Context("cluster type shoot", func() {
 			BeforeEach(func() {
-				vpa = New(c, namespace, sm, Values{
-					ClusterType:              component.ClusterTypeShoot,
-					Enabled:                  true,
-					SecretNameServerCA:       secretNameCA,
-					RuntimeKubernetesVersion: runtimeKubernetesVersion,
-					AdmissionController:      valuesAdmissionController,
-					Recommender:              valuesRecommender,
-					Updater:                  valuesUpdater,
-				})
+				vpa = vpaFor(component.ClusterTypeShoot, false)
 				managedResourceName = "shoot-core-vpa"
 			})
 
+			Context("when deploying ServiceMonitors", func() {
+				var serviceMonitorExpected = &monitoringv1.ServiceMonitor{}
+
+				BeforeEach(func() {
+					serviceMonitorAdmissionController.TypeMeta = metav1.TypeMeta{}
+					serviceMonitorAdmissionController.ResourceVersion = "1"
+					serviceMonitorAdmissionController.ObjectMeta.Name = "shoot-vpa-admission-controller"
+					serviceMonitorAdmissionController.ObjectMeta.Labels = map[string]string{
+						"prometheus": "shoot",
+					}
+
+					serviceMonitorExpected = serviceMonitorRecommenderFor(component.ClusterTypeShoot)
+					serviceMonitorExpected.ResourceVersion = "1"
+				})
+
+				Context("in a garden cluster", func() {
+					BeforeEach(func() {
+						vpa = vpaFor(component.ClusterTypeShoot, true)
+						Expect(vpa.Deploy(ctx)).To(Succeed())
+					})
+
+					It("should attach `prometheus=shoot` to vpa-admission-controller ServiceMonitor", func() {
```

Comment:
Let's keep the same description as the tests above:
```suggestion
					It("should label vpa-admission-controller ServiceMonitor with `prometheus=shoot`", func() {
```

Reply:
Thanks for pointing this mismatched wording. I updated the `shoot` tests to have their titles matching the previously defined cases.